Sources/Application/MainCoordinator.swift

```diff
@@ -57,19 +57,23 @@ final class MainCoordinator: NSObject,
             return
         }
 
-        do {
-            try sessionManager.resumeSession()
-            updateToken()
-            windowManager.hideUnlockWindow()
-        } catch {
-            switch error {
-            case is JWTVerifierError,
-                SecureStoreError.unableToRetrieveFromUserDefaults,
-                SecureStoreError.cantInitialiseData,
-                SecureStoreError.cantRetrieveKey:
-                fullLogin(loginError: error)
-            default:
-                print("Token retrival error: \(error)")
+        Task {
+            await MainActor.run {
```

Comment:
I'm not sure this is the right approach. I think we should make the `MainCoordinator` a MainActor, rather than wrapping things in a task. Coordinators should always be main actors.

Reply:
Yeah, I was hoping that there was some other way to do this, I did have a play around with marking this as `MainActor` and it didn't seem to solve the app crashing after 15~ seconds with the local auth prompt displayed on the Unlock screen. Maybe one to talk about in-person